harness/determined/core/_train.py

```diff
@@ -68,6 +68,18 @@ def _get_last_validation(self) -> Optional[int]:
         logger.debug(f"_get_last_validation() -> {steps_completed}")
         return steps_completed
 
+    def log_metadata(self, metadata: Dict[str, Any]) -> None:
+        """
+        Log run metadata to the Determined master.
+
+        The metadata is a dictionary of key-value pairs that can be used for analysis,
+        post-processing, or debugging.
```

Comment:
should add some info here about how adding a key might merge/overwrite existing metadata

Reply:
Absolutely, the expected behavior will be that additional calls will overwrite existing metadata.

I guess one way to "delete" metadata while if using the Core API would be to make a request with an empty dictionary.